cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/SideEffects.qll

```diff
@@ -45,7 +55,7 @@ private predicate hasDefaultSideEffect(Call call, ParameterIndex i, boolean buff
       ) and
       (
         isWrite = true and
-        not call.getTarget().getParameter(i).getType().isDeeplyConstBelow()
+        not isConstPointerLike(call.getTarget().getParameter(i).getUnderlyingType())
```

Comment:
Doesn't `isDeeplyConstBelow` require that, if you have multiple levels of indirection, that all levels are `const`? If so, that doesn't seem to be what `isConstPointerLike()` does.

Reply:
Hm, yes. Good point. I'll fix that (and probably add some more tests).

Reply:
Fixed in 27586d77f87a51d42204745b8c7f1351cbf60131. I did a force push to make the commits come in a reviewer-friendly order.

Reply:
👍